pkg/cachemanager/cachemanager.go

```diff
@@ -0,0 +1,370 @@
+package cachemanager
+
+import (
+	"context"
+	"fmt"
+	"sync"
+	"time"
+
+	"github.com/open-policy-agent/gatekeeper/v3/pkg/cachemanager/aggregator"
+	"github.com/open-policy-agent/gatekeeper/v3/pkg/controller/config/process"
+	"github.com/open-policy-agent/gatekeeper/v3/pkg/metrics"
+	"github.com/open-policy-agent/gatekeeper/v3/pkg/readiness"
+	"github.com/open-policy-agent/gatekeeper/v3/pkg/syncutil"
+	"github.com/open-policy-agent/gatekeeper/v3/pkg/target"
+	"github.com/open-policy-agent/gatekeeper/v3/pkg/watch"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/apimachinery/pkg/util/wait"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	logf "sigs.k8s.io/controller-runtime/pkg/log"
+)
+
+var (
+	log     = logf.Log.WithName("cache-manager")
+	backoff = wait.Backoff{
+		Duration: time.Second,
+		Factor:   2,
+		Jitter:   0.1,
+		Steps:    3,
+	}
+)
+
+type Config struct {
+	Opa              syncutil.OpaDataClient
+	SyncMetricsCache *syncutil.MetricsCache
+	Tracker          *readiness.Tracker
+	ProcessExcluder  *process.Excluder
+	Registrar        *watch.Registrar
+	WatchedSet       *watch.Set
+	GVKAggregator    *aggregator.GVKAgreggator
+	Reader           client.Reader
+}
+
+type CacheManager struct {
+	watchedSet            *watch.Set
+	processExcluder       *process.Excluder
+	gvksToSync            *aggregator.GVKAgreggator
+	needToList            bool
+	gvksToDeleteFromCache *watch.Set
+	excluderChanged       bool
+	// mu guards access to any of the fields above
+	mu sync.RWMutex
+
+	opa                        syncutil.OpaDataClient
+	syncMetricsCache           *syncutil.MetricsCache
+	tracker                    *readiness.Tracker
+	registrar                  *watch.Registrar
+	backgroundManagementTicker time.Ticker
+	reader                     client.Reader
+
+	// stopChan is used to stop any list operations still in progress
+	stopChan chan bool
```

Comment:
We should make the name more specific. `stopChan` implies this stops the entire resource.

Reply:
renamed to `relistStopChan`.